src/components/fast_clock.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * @file fast_clock.c
+ * @brief Loconet clock implementation
+ *
+ * \copyright Copyright 2017 /Dev. All rights reserved.
+ * \license This project is released under MIT license.
+ *
+ * @author Jan Martijn van der Werf
+ * @see fast_clock.h
+ */
+
+#include "fast_clock.h"
+
+// ----------------------------------------------------------------------------
+// Prototypes
+void fast_clock_handle_update_(FAST_CLOCK_TIME_Type time);
+
+// ----------------------------------------------------------------------------
+// this is the function that any system that wants to react on the clock should
+// implement.
+//
+__attribute__ ((weak, alias("fast_clock_handle_update_")))
+  void fast_clock_handle_update(FAST_CLOCK_TIME_Type time);
+
+// ----------------------------------------------------------------------------
+// set time initially to sunday 00:00:00.0000
+FAST_CLOCK_TIME_Type current_time = {0, 0, 0, 0};
+
+// ----------------------------------------------------------------------------
+typedef struct {
+  bool master;
+  uint8_t id1;
+  uint8_t id2;
+  uint16_t intermessage_delay;
+  uint8_t rate;
+  uint8_t millisecond;
+} FAST_CLOCK_STATUS_Type;
+
+FAST_CLOCK_STATUS_Type fast_clock_status = {0, 0, 0, 0, 1, 0};
+
+uint16_t fast_clock_current_intermessage_delay = 0;
+
+// ----------------------------------------------------------------------------
+// The clock is defined on div8. That means that the clock is triggered
+// every 50ms.
```

Comment:
The clock is triggered faster, every 1us. But with the delay defined below you get to 50ms. Please reflect that in the comment.